cpp/src/arrow/dataset/test_util.h

```diff
@@ -139,5 +139,45 @@ std::unique_ptr<GeneratedRecordBatch<Gen>> MakeGeneratedRecordBatch(
   return internal::make_unique<GeneratedRecordBatch<Gen>>(schema, std::forward<Gen>(gen));
 }
 
+/// \brief A dummy FileFormat implementation
+class DummyFileFormat : public FileFormat {
+ public:
+  std::string name() const override { return "dummy"; }
+
+  /// \brief Return true if the given file extension
+  bool IsKnownExtension(const std::string& ext) const override { return ext == name(); }
+
+  /// \brief Open a file for scanning (always returns an empty iterator)
+  Status ScanFile(const FileSource& source, std::shared_ptr<ScanOptions> scan_options,
+                  std::shared_ptr<ScanContext> scan_context,
+                  std::unique_ptr<ScanTaskIterator>* out) const override {
+    *out = internal::make_unique<EmptyIterator<std::unique_ptr<ScanTask>>>();
+    return Status::OK();
+  }
+
+  Status MakeFragment(const FileSource& location, std::shared_ptr<ScanOptions> opts,
+                      std::unique_ptr<FileBasedDataFragment>* out) override;
```

Comment:
Is it ok to `override` with a slightly different signature? (I suspect the base declaration doesn't use `FileBasedDataFragment`).

Reply:
It does use FileBasedDataFragment; I wasn't intending that a FileFormat would produce any other kind of data fragment. I'll change the signature for ease of later extension to non-path `FileSource`s